scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/spel/Typer.scala

```diff
@@ -129,24 +153,35 @@ private[spel] class Typer(
       current: TypingContext
   ): NodeTypingResult = {
 
-    def toNodeResult(typ: TypingResult) = current.toResult(TypedNode(node, TypingResultWithContext(typ)))
+    implicit class TypingResultOps(typ: TypingResult) {
```

Comment:
There are a lot of changes in this file, but basically:
- I changed prefix notation to postfix notation for some methods to reduce the level of braces nesting (I had to add one more level (`withPositionRange`).
- I made some private class methods local for `typeNode`.

Thanks to these changes, it was easy to add information about the position range to every error.